src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs

```diff
@@ -153,6 +153,16 @@ private static void SetElement(XmlDocument document, XmlNode node, string name,
 
             foreach (DependencyDetail itemToUpdate in itemsToUpdate)
             {
+                if (string.IsNullOrEmpty(itemToUpdate.Version) ||
+                    string.IsNullOrEmpty(itemToUpdate.Name) ||
+                    string.IsNullOrEmpty(itemToUpdate.Commit) ||
+                    string.IsNullOrEmpty(itemToUpdate.RepoUri))
+                {
+                    _logger.LogError("Either the name, version, commit or repo uri of a dependency in " +
+                        $"repo '{repoUri}' and branch '{branch}' was empty...skipping.");
+                    continue;
```

Comment:
This should probably throw instead.  This is an error that is introducing incorrect data to the model.  We shouldn't just skip the update.